codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -427,16 +420,16 @@ public void error(Throwable cause) {
 
         @Override
         public boolean write(int allowedBytes) {
-            if (data == null) {
-                return false;
-            }
-            if (allowedBytes == 0 && size() != 0) {
-                // No point writing an empty DATA frame, wait for a bigger allowance.
-                return false;
-            }
-            int maxFrameSize = frameWriter().configuration().frameSizePolicy().maxFrameSize();
+            int bytesWritten = 0;
             try {
-                int bytesWritten = 0;
+                if (data == null) {
+                    return false;
+                }
+                if (allowedBytes == 0 && size() != 0) {
```

Comment:
I know you didn't introduce this but we use `size()` and `size` in this method.  Lets be consistent.